backend/src/bundles/workouts/workout.controller.ts

```diff
@@ -0,0 +1,353 @@
+     *                     enum:
+     *                        - cycling
+     *                        - running
+     *                        - walking
+     *      responses:
+     *        200:
+     *          description: Successful operation
+     *          content:
+     *            application/json:
+     *              schema:
+     *                 type: object
+     *                 $ref: '#/components/schemas/Workout'
+     *        400:
+     *          description: Failed operation
+     *          content:
+     *              application/json:
+     *                  schema:
+     *                      type: object
+     *                      $ref: '#/components/schemas/Error'
+     */
+    private async create(
+        options: ApiHandlerOptions<{
+            body: CreateWorkoutRequestDto;
+        }>,
+    ): Promise<ApiHandlerResponse> {
+        return {
+            status: HttpCode.OK,
+            payload: await this.workoutService.create({
+                ...options.body,
+                userId: (options.user as UserAuthResponseDto).id,
+            }),
+        };
+    }
+    /**
+     * @swagger
+     * /api/v1/workouts/{id}:
+     *    put:
+     *      parameters:
+     *      - in: path
+     *        name: id
+     *        required: true
+     *      tags:
+     *       - Workouts
+     *      description: Updates workout
+     *      security:
+     *        - bearerAuth: []
+     *      requestBody:
+     *        description: Data for workout
+     *        required: true
+     *        content:
+     *          application/json:
+     *            schema:
+     *              type: object
+     *              properties:
+     *                 activity:
+     *                     type: string
+     *                     enum:
+     *                        - cycling
+     *                        - running
+     *                        - walking
+     *                 steps:
+     *                     type: number
+     *                     minimum: 0
+     *                 duration:
+     *                     type: number
+     *                     minimum: 0
+     *                 kilocalories:
+     *                     type: number
+     *                     minimum: 0
+     *      responses:
+     *        200:
+     *          description: Successful operation
+     *          content:
+     *            application/json:
+     *              schema:
+     *                 type: object
+     *                 $ref: '#/components/schemas/Workout'
+     *        400:
+     *          description: Failed operation
+     *          content:
+     *              application/json:
+     *                  schema:
+     *                      type: object
+     *                      $ref: '#/components/schemas/Error'
+     */
+    private async update(
+        options: ApiHandlerOptions<{
+            body: WorkoutRequestDto;
+            params: { id: string };
```

Comment:
```suggestion
            body: UpdateWorkoutRequestDto;
            params: EntityIdParamsDto
```
Also lets add validation schema for parameters in all requests